bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/ManagedMetadataProviderImpl.java

```diff
@@ -76,4 +81,30 @@ public void removeItemMetadata(String name) {
         logger.debug("Removing all metadata for item {}", name);
         getAll().stream().filter(MetadataPredicates.ofItem(name)).map(Metadata::getUID).forEach(this::remove);
     }
+
+    @Override
+    public Collection<Metadata> getAll() {
+        return super.getAll().stream().map(this::normalizeMetadata).collect(Collectors.toUnmodifiableList());
+    }
+
+    @Override
+    public @Nullable Metadata get(MetadataKey key) {
+        Metadata metadata = super.get(key);
+        if (metadata != null) {
+            return normalizeMetadata(metadata);
+        }
+        return null;
+    }
+
+    private Metadata normalizeMetadata(Metadata metadata) {
+        return new Metadata(metadata.getUID(), metadata.getValue(), metadata.getConfiguration().entrySet().stream()
+                .map(this::normalizeConfigEntry).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+    }
+
+    private Map.Entry<String, Object> normalizeConfigEntry(Map.Entry<String, Object> entry) {
+        if (entry.getValue() instanceof Number) {
+            return Map.entry(entry.getKey(), new BigDecimal(entry.getValue().toString()));
```

Comment:
I'm not sure if it is relevant, but in case the number is an int, we [here](https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/internal/normalization/IntNormalizer.java#L51) also set the scale to 0 to make sure that it is not considered a floating point value. Wdyt, would this make sense here as well?

Reply:
I'm not sure if it is necessary, but it doesn't hurt. Implemented.